src/XHRConnection.js

```diff
@@ -35,18 +35,24 @@ export default class XHRConnection extends AbstractConnection {
         Accept: "application/json"
       },
       responseType = "json",
-      timeout = 0
+      timeout = 0,
+      listeners = [],
```

Comment:
This seems odd to me, having addListeners introduced in the "parent" and making the setting and handling of listeners in the child. Can we "just" put them into the super call and move all of this logic on the bottom of the constructor in the abstract connection?

Reply:
`addListerners` isnt introduced in `AbstractConnection`?!